interval/interval.go

```diff
@@ -12,6 +12,7 @@ import (
 	"golang.org/x/exp/slices"
 )
 
+// Interval is a type interface for any genomic datatype that has chromosomal coordinate info (chromosome, start, end)
```

Comment:
```suggestion
// Interval is a type interface for any genomic datatype that has chromosomal coordinate info (chromosome, start, end)
// Note that the methods must return positions for a 0-base half-closed interval [start, end)
```

We should also give some specifics about the implementation since this is something users will need to interface with if they want to use interval for unsupported file types.

Reply:
To help people out we could say something like, "In order to satisfy this interface, a type must have GetChrom(), GetChromStart(), and GetChromEnd() methods."  Would that be good?